### ISSUE TITLE
Antox crash repeatedly

### ISSUE THREAD

**Issue by ghost** [Antox/Antox#372]
I've encountered this issue since many months ago. Now I've reported this problem to here.

Antox v0.24.422 (Fdroid)
Non-rooted(unrootable), Android 5.1.1

Step:
1. Install Antox as normal.
2. Open it. Type "default" into login menu, un-tick ToxMe checkbox, and tap login.
3. Instant crash.
4. Tried to open Antox, instant crash.
5. Goto 4.

Info:
I've used this script[1] to clean up my phone. Almost all apps which I've installed via Fdroid works, except this one.
Any idea which I need to re-enable to use Antox?

[1] https://www.autistici.org/a2af37vnxe44tcgo/data/memo/m/andr_5xbwlist.txt

**Comment by GitHubHatesTor** [comment]
Another step:
1. Setting > Apps > Antox > Force stop.
2. Tried to open, still instant crash.
3. Do Step 1 again, this time, tap "clear data".
4. Now Antox opened.
5. In login menu, tap = (burger) > Settings. NOTHING HAPPENED(expected result: show settings and let user to set proxy. I use proxy, without it app can't connect to the internet!).
6. Input "test" to profile name, make sure Toxme is unchecked, and tap register. Crash.

Why?

**Comment by licaon-kter** [comment]
Duplicate of #324 with #325 with #326 with #327 and #328. I guess.

**Comment by GitHubHatesTor** [comment]
Not a duplicate because I couldn't open a settings to set a proxy for Antox.
And if I did, Antox will still crash, like I described above.

**Comment by licaon-kter** [comment]
Read them all, together: it will crash ( #326 ) since it can't download via clearnet ( #328 ) that file ( #327 ).

**Comment by GitHubHatesTor** [comment]
@licaon-kter

Yes I know what clearnet is. And yes I'm a tor lover.

> Crash if Antox can't download nodefile.json #326

Wow, Antox crash itself if it can't download ONLY ONE FILE!?
Why not show an error "Unable to connect" instead of crashing without explanation?

**Comment by licaon-kter** [comment]
So, I guess you can close this since might be a duplicate of #326 
Do get a log of the crash to confirm this.
Or let it connect via clearnet and see it it does not crash then.

**Comment by GitHubHatesTor** [comment]
Same result... v0.24.424

**Comment by GitHubHatesTor** [comment]
NetGuard is not great; most of them are blocked by PAYWALL. I recommend "NoRoot Data Firewall" instead of it. https://www.apk4fun.com/apk/55628/

**Comment by licaon-kter** [comment]
NetGuard is great, development costs (time). Hope that other closed source app helps you track these connections.

**Comment by zoff99** [comment]
a nodefile is included but never used.
as a temporary solution you can use this development version:
https://github.com/zoff99/Antox/tree/zoff99/Antox_v0.25.1
build is now failing but will be ok soon.
if it works with this, you can make a PR

**Comment by githubbantor** [comment]
UPdated to latest and tried again, still no lock. Cay I cry?

**Comment by zoff99** [comment]
@githubbantor where did you get that latest version from?

**Comment by githubbantor** [comment]
@zoff99 from FDroid with tox repository of course

**Comment by zoff99** [comment]
as i said, can you try the version from my link, and see if that version works?

**Comment by hasufell** [comment]
Takes long to start, freezes occasionally, crashes occasionally when trying to write something. Unusable.

**Comment by zoff99** [comment]
@hasufell what version did you use?

**Comment by hasufell** [comment]
@zoff99 0.25.458

**Comment by zoff99** [comment]
can you try my link above, and see if there is any difference in behaviour?
to see if some code should be merged

**Comment by hasufell** [comment]
@zoff99 indeed that see to work better

**Comment by zoff99** [comment]
@hasufell sounds good. can you give more details for maintainers to fix the issues?

**Comment by subliun** [comment]
Fixed
